pkg/endpoint/endpoint.go

```diff
@@ -182,6 +189,74 @@ func getCiliumClient() (cilium_client_v2.CiliumV2Interface, error) {
 	return k8sClient.CiliumV2(), nil
 }
 
+// runK8sCiliumEndpointSyncGC starts the node-singleton sweeper for
+// CiliumEndpoint objects where the managing node is no longer running
+func runK8sCiliumEndpointSyncGC() {
+	controllerName := fmt.Sprintf("sync-to-k8s-ciliumendpoint-gc (%v)", node.GetName())
+	scopedLog := log.WithField("controller", controllerName)
+
+	if !k8s.IsEnabled() {
+		scopedLog.WithField("name", controllerName).Debug("Not starting controller because k8s is disabled")
+		return
+	}
+
+	ciliumClient, err := getCiliumClient()
+	if err != nil {
+		scopedLog.WithError(err).Error("Not starting controller because unable to get cilium k8s client")
+		return
+	}
+
+	runThrottler := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	// this dummy manager is needed only to add this controller to the global list
+	controller.NewManager().UpdateController(controllerName,
```

Comment:
What about running the controller with a runinterval of 1 second, and pooling every second for 1 minute to detect if `k8s` is enabled? if it returns false after 60 seconds, we kill the controller

Reply:
It could work. I should check if we can change the controller interval while it's running or not. Otherwise, I'll try to do something more elegant :/